functions/vshn-postgres-func/alerting.go

```diff
@@ -0,0 +1,194 @@
+		}
+
+		log.Info("Found an AlertmanagerConfigTemplate, deploying...")
+
+		err := deployAlertmanagerFromTemplate(comp, iof)
+		if err != nil {
+			return comp, err
+		}
+	}
+
+	if monitoringSpec.AlertmanagerConfigSecretRef != "" {
+		refName := comp.Spec.Parameters.Monitoring.AlertmanagerConfigSecretRef
+		log.Info("Found an AlertmanagerConfigSecretRef, deploying...", "refName", refName)
+
+		err := deploySecretRef(comp, iof)
+		if err != nil {
+			return comp, err
+		}
+	}
+
+	return comp, nil
+}
+
+func deployAlertmanagerFromRef(comp *vshnv1.VSHNPostgreSQL, iof *runtime.Runtime) error {
+	ac := &alertmanagerv1alpha1.AlertmanagerConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "postgresql-alertmanagerconfig",
+			Namespace: comp.Status.InstanceNamespace,
+		},
+	}
+
+	xkobj, err := addObjectToXKube(comp, "-alertmanagerconfig", ac)
+	if err != nil {
+		return err
+	}
+
+	xkobj.Spec.References = []xkube.Reference{
+		{
+			PatchesFrom: &xkube.PatchesFrom{
+				DependsOn: xkube.DependsOn{
+					APIVersion: "monitoring.coreos.com/v1alpha1",
+					Kind:       "AlertmanagerConfig",
+					Namespace:  comp.ObjectMeta.Labels["crossplane.io/claim-namespace"],
+					Name:       comp.Spec.Parameters.Monitoring.AlertmanagerConfigRef,
+				},
+				FieldPath: pointer.String("spec"),
+			},
+			ToFieldPath: pointer.String("spec"),
+		},
+	}
+
+	err = iof.PutManagedRessource(xkobj)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func deployAlertmanagerFromTemplate(comp *vshnv1.VSHNPostgreSQL, iof *runtime.Runtime) error {
+	ac := &alertmanagerv1alpha1.AlertmanagerConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "postgresql-alertmanagerconfig",
+			Namespace: comp.Status.InstanceNamespace,
+		},
+		Spec: *comp.Spec.Parameters.Monitoring.AlertmanagerConfigSpecTemplate,
+	}
+
+	xkobj, err := addObjectToXKube(comp, "-alertmanagerconfig", ac)
+	if err != nil {
+		return err
+	}
+
+	return iof.PutManagedRessource(xkobj)
+}
+
+func deploySecretRef(comp *vshnv1.VSHNPostgreSQL, iof *runtime.Runtime) error {
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "postgresql-alertmanagerconfigsecret",
+			Namespace: comp.Status.InstanceNamespace,
+		},
+	}
+	xkobj, err := addObjectToXKube(comp, "-alertmanagerconfigsecret", secret)
+	if err != nil {
+		return err
+	}
+
+	xkobj.ObjectMeta.Name = comp.Name + "-alertmanagerconfigsecret"
+	xkobj.Spec.References = []xkube.Reference{
+		{
+			PatchesFrom: &xkube.PatchesFrom{
+				DependsOn: xkube.DependsOn{
+					APIVersion: "v1",
+					Kind:       "Secret",
+					Namespace:  comp.ObjectMeta.Labels["crossplane.io/claim-namespace"],
+					Name:       comp.Spec.Parameters.Monitoring.AlertmanagerConfigSecretRef,
+				},
+				FieldPath: pointer.String("data"),
+			},
+			ToFieldPath: pointer.String("data"),
+		},
+	}
+
+	return iof.PutManagedRessource(xkobj)
+}
+
+func addObjectToXKube(comp *vshnv1.VSHNPostgreSQL, namesuffix string, obj client.Object) (*xkube.Object, error) {
```

Comment:
Nitpick: This can be written generically and IMO it's very likely that we'll need this again. So I'd probably add this to the runtime.

Reply:
Yeah, through all the rebasing I forgot, that we actually have such a function the library already, but not public. I'll probably just use that one then.